streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -234,10 +256,10 @@ private ClientResponse sendRequest(final ClientRequest clientRequest) {
     /**
      * Fetch the metadata for all topics
      */
-    public Collection<MetadataResponse.TopicMetadata> fetchTopicsMetadata() {
+    public MetadataResponse fetchMetadata() {
 
         final ClientRequest clientRequest = kafkaClient.newClientRequest(
-            getBrokerId(),
+            getAnyReadyBrokerId(),
```

Comment:
I don't think so. The metadata can be received from any broker. That's how one bootstraps, e.g., with a bootstrap list, which might not include the controller. cc @ijuma